File: javaapp/src/main/java/com/bluthlee/javaapp/MainActivity.java
package com.bluthlee.javaapp;

import android.os.Environment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.bluthlee.javalibrary.AudioRecorder;
import com.bluthlee.javalibrary.DragAudioRecorder;
import com.bluthlee.javalibrary.FloatAudioRecorder;
import com.bluthlee.javalibrary.ui.recorderWidget.FloatRecorderWidget;

public class MainActivity extends AppCompatActivity {

    private Button button1;
    private Button button2;
    private FloatAudioRecorder floatAudioRecorder;

    private String dirPath = Environment.getExternalStorageDirectory().getAbsolutePath() + "/audio_recorder";
    private String dragDirPath = dirPath + "/drag";
    private String floatDirPath = dirPath + "/float";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);

        //使用DragAudioRecorder
        DragAudioRecorder.Builder
                //required
                .setAttachedRecorderButton(button1)

                //optional
                .setDirPath(dragDirPath)
                .setFileName(System.currentTimeMillis() + "_drag")
                .setOnFinishListener(new AudioRecorder.OnFinishListener() {
                    @Override
                    public void onFinish(String filePath) {
                        Toast.makeText(MainActivity.this, filePath, Toast.LENGTH_SHORT).show();
                    }
                })

                //build and initialize
                .build();

        //使用FloatAudioRecorder
        floatAudioRecorder = FloatAudioRecorder.Builder
                //required
                .setContext(this)

                //optional
                .setDirPath(floatDirPath)
                .setFileName(System.currentTimeMillis() + "")
                .setOnFinishListener(new AudioRecorder.OnFinishListener() {
                    @Override
                    public void onFinish(String filePath) {
                        Toast.makeText(MainActivity.this, filePath, Toast.LENGTH_SHORT).show();
                    }
                })

                //build and initialize
                .build();
    }

    public void click(View view) {
        if (!floatAudioRecorder.isShowing()) {
            floatAudioRecorder.show();
        } else {
            floatAudioRecorder.cancel();
        }
    }

}

File: app/src/main/java/com/bluthlee/audiorecorder/MainActivity.kt
package com.bluthlee.audiorecorder

import android.os.Bundle
import android.os.Environment
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import com.bluthlee.library.AudioRecorder
import com.bluthlee.library.DragAudioRecorder
import com.bluthlee.library.FloatAudioRecorder
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by LC on 2017/8/24.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var floatAudioRecorder: FloatAudioRecorder

    private val dirPath = Environment.getExternalStorageDirectory().absolutePath + "/audio_recorder"
    private val dragDirPath = dirPath + "/drag"
    private val floatDirPath = dirPath + "/float"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //使用DragAudioRecorder
        DragAudioRecorder.Builder
                //required
                .setAttachedRecorderButton(button1)

                //optional
                .setDirPath(dragDirPath)
                .setFileName(System.currentTimeMillis().toString() + "_drag")
                .setOnFinishListener(object : AudioRecorder.OnFinishListener {
                    override fun onFinish(filePath: String) {
                        Toast.makeText(this@MainActivity, filePath, Toast.LENGTH_SHORT).show()
                    }
                })

                //build and initialize
                .build()

        //使用FloatAudioRecorder
        floatAudioRecorder = FloatAudioRecorder.Builder
                //required
                .setContext(this)

                //optional
                .setDirPath(floatDirPath)
                .setFileName(System.currentTimeMillis().toString() + "_float")
                .setOnFinishListener(object : AudioRecorder.OnFinishListener {
                    override fun onFinish(filePath: String) {
                        Toast.makeText(this@MainActivity, filePath, Toast.LENGTH_SHORT).show()
                    }
                })

                //build and initialize
                .build()
    }

    fun click(view: View) {
        if (!floatAudioRecorder.isShowing()) {
            floatAudioRecorder.show()
        } else {
            floatAudioRecorder.cancel()
        }
    }

}